src/lamplib/src/genny/tasks/compile.py

```diff
@@ -32,6 +33,85 @@ def _sanitizer_flags(sanitizer: str, genny_repo_root: str):
     raise ValueError("Unknown sanitizer {}".format(sanitizer))
 
 
+class DistroDetectionError(Exception):
+    pass
+
+
+def _detect_distro_ubuntu(machine, freedesktop_version):
+    if freedesktop_version == "22.04":
+        if machine == "aarch64":
+            return "ubuntu2204_arm64"
+        elif machine == "x86_64":
+            return "ubuntu2204"
+        else:
+            raise DistroDetectionError(f"Invalid machine type for Ubuntu 22.04: {machine}")
+    elif freedesktop_version == "20.04":
+        if machine == "aarch64":
+            return "ubuntu2004_arm64"
+        elif machine == "x86_64":
+            return "ubuntu2004"
+        else:
+            raise DistroDetectionError(f"Invalid machine type for Ubunutu 20.04: {machine}")
+    elif freedesktop_version == "18.04":
+        if machine == "x86_64":
+            return "ubuntu1804"
+        else:
+            raise DistroDetectionError(f"Invalid machine type for Ubuntu 18.04: {machine}")
+    else:
+        raise DistroDetectionError(f"Invalid version for Ubuntu: {freedesktop_version}")
+
+
+def _detect_distro_rhel(machine, freedesktop_version):
+    # We only distinguish between major versions of RHEL, but they include minor versions in their version_id
+    if freedesktop_version.startswith("7."):
+        if machine == "x86_64":
+            return "rhel70"
+        else:
+            raise DistroDetectionError(f"Invalid machine type for RHEL 7.x: {machine}")
+    elif freedesktop_version.startswith("8."):
+        if machine == "x86_64":
+            return "rhel8"
+        else:
+            raise DistroDetectionError(f"Invalid machine type for RHEL 8.x: {machine}")
+
+
+def _detect_distro_amazon(machine, freedesktop_version):
+    if freedesktop_version == "2":
+        if machine == "aarch64":
+            return "amazon2_arm64"
+        elif machine == "x86_64":
+            return "amazon2"
+        else:
+            raise DistroDetectionError(f"Invalid machine type for Amazon 2: {machine}")
+    else:
+        raise DistroDetectionError(f"Invalid version for Amazon Linux: {freedesktop_version}")
+
+
+def detect_distro():
+    SLOG.info(f"Distro not specified. Detecting distro.")
+
+    system = platform.system()
+    machine = platform.machine()
```

Comment:
Nit: Maybe can move line 94 to between line 100 & 101, so that the `machine` variable is closer to where it's actually used?

Reply:
Fixed